yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -372,11 +381,29 @@ private[spark] class Client(
       sparkConf.set(CONF_SPARK_YARN_SECONDARY_JARS, cachedSecondaryJarLinks.mkString(","))
     }
 
+    if (args.primaryPyFile != null) {
+      distribute(args.primaryPyFile)
```

Comment:
Don't you need the file around when you run code in the executors? How is the code shipped to executors otherwise?

I can try it out, but it feels weird to have used code only in the AM.